Parse/ParseUser.cs

```diff
@@ -558,6 +558,7 @@ public class ParseUser : ParseObject {
           authData = AuthData = new Dictionary<string, IDictionary<string, object>>();
         }
         authData[authType] = data;
+        AuthData = authData;
```

Comment:
This is needed, as we were previously relying on the dictionary magically being updated.